app/lib/widgets/cards/productCard/productCard.dart

```diff
@@ -0,0 +1,180 @@
+import 'package:flex_ui/tokens/colors.dart';
+import 'package:flex_ui/utils/extension.dart';
+import 'package:flex_ui/widgets/cards/productCard/orientations/content_product_card.dart';
+import 'package:flex_ui/widgets/cards/productCard/shared/right_bottom_icon_button.dart';
+import 'package:flex_ui/widgets/cards/productCard/shared/left_top_icon.dart';
+import 'package:flutter/material.dart';
+import 'package:widgetbook/widgetbook.dart';
+import 'package:widgetbook_annotation/widgetbook_annotation.dart' as widgetbook;
+
+class FlexProductCard extends StatelessWidget {
+  FlexProductCard({
+    super.key,
+    required this.productName,
+    this.productReference,
+    this.imageUrl,
+    required this.price,
+    this.oldPrice,
+    required this.currency,
+    this.notation,
+    this.displayLeftIcon = false,
+    this.leftIconLabel,
+    this.leftIconSemanticsLabel,
+    this.leftIconBackgroundColor,
+    this.leftIconTextColor,
+    this.displayRightIcon = false,
+    this.rightIcon,
+    this.rightIconButtonStyle,
+    this.rightIconSemanticsLabel,
+    this.onPressedRightIcon,
+    this.isSaved = false,
+    this.isAvailable = true,
+    this.isLandscape = false,
+  })  : assert(
+          displayLeftIcon == false || leftIconLabel.isNotBlank(),
+          'You must provide a leftIconLabel when displayLeftIcon is true',
+        ),
+        assert(
+          displayRightIcon == false ||
+              (rightIcon != null && onPressedRightIcon != null),
+          'You must provide a rightIcon and onPressedRightIcon when displayRightIcon is true',
+        );
+
+  final String productName;
+  final String? productReference;
+  final String? imageUrl;
+  final double price;
+  final double? oldPrice;
+  final String currency;
+  final int? notation;
+  final bool displayLeftIcon;
+  final String? leftIconLabel;
+  final String? leftIconSemanticsLabel;
+  final Color? leftIconBackgroundColor;
+  final Color? leftIconTextColor;
+  final bool displayRightIcon;
+  final Icon? rightIcon;
+  final ButtonStyle? rightIconButtonStyle;
+  final String? rightIconSemanticsLabel;
+  final VoidCallback? onPressedRightIcon;
+  final bool isSaved;
+  final bool isAvailable;
+  final bool isLandscape;
+
+  @override
+  Widget build(BuildContext context) {
+    return ConstrainedBox(
+      constraints: BoxConstraints(
+          maxWidth: isLandscape ? double.infinity : 400,
+          minWidth: 200,
+          maxHeight: isLandscape ? 400 : double.infinity,
+          minHeight: 400),
```

Comment:
Good start and works for first version, but magic numbers 😅

Reply:
aha yes you right I choose the easiest solution  
If we put just some constraints on min width/height, should be ok ? I think it's  up to the developer using the widget to set its size from his parent. We just need to be ensured that the widget don't be too small 

```
constraints: BoxConstraints(
        minWidth: MediaQuery.of(context).size.width * 0.25,
        minHeight: MediaQuery.of(context).size.width * 0.25,
      ),
```
And should we expose widget's height/width to the developer ?

Reply:
Let's attempt some usage of this widget without exposing width and height